File: FrontEnd/src/models/post.ts
import type { ImagesPreview } from './file';
import type { User } from './user';
import {Timestamp} from "firebase/firestore";

export type Post = {
  id: string;
  user: User  ;
  content: string | null;
  images: ImagesPreview | null;
  parent: { id: string; username: string } | null;
  createdBy: string;
  createdAt: string;
  updatedAt: string | null;
  status: boolean;
  tagUserPosts: [];
  userPostLikes: [];
  priorityScore: number;
  userReplies: number;
  totalLikes: number;
  totalComments: number;
};

export type TweetWithUser = Post & { user: User };

File: BackEnd/realtime-service/src/main/java/com/mytech/realtimeservice/models/Post.java
package com.mytech.realtimeservice.models;
import com.mytech.realtimeservice.models.users.User;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Data
@Document
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Post implements Serializable {
    @Id
    private String id;
    private User user;
    private String content;
    private LocalDateTime createdAt;
    private String createdBy;
    private LocalDateTime updatedAt;
    private List<MediaObject> mediaUrl;
    private boolean status;
    private List<User> tagUserPosts;
    private double priorityScore;
    private List<User> userPostLikes = new ArrayList<>();
    private int userReplies;
    private int totalLikes;
    private int totalComments;
    private int totalShares;
    private LocalDateTime lastInteractionAt;
    private boolean leastPrioritized;

    @Transient
    private double engagementScore;
}
